diff --git a/third_party/libtiff/tif_ojpeg.c b/third_party/libtiff/tif_ojpeg.c
index cc5449cd6..45152be98 100644
--- a/third_party/libtiff/tif_ojpeg.c
+++ b/third_party/libtiff/tif_ojpeg.c
@@ -1918,7 +1918,10 @@ OJPEGReadHeaderInfoSecTablesAcTable(TIFF* tif)
                                rb[sizeof(uint32)+5+n]=o[n];
                        p=TIFFReadFile(tif,&(rb[sizeof(uint32)+21]),q);
                        if (p!=q)
+                       {
+                               _TIFFfree(rb);
                                return(0);
+                       }
                        sp->actable[m]=rb;
                        sp->sos_tda[m]=(sp->sos_tda[m]|m);
                }
